app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,346 @@
+                            onClick = {
+                                navController.navigate(NavGraphDirections.actionGlobalTurnOnSync())
+                            }
+                        )
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_sign_in),
+                        contentDescription = null,
+                        tint = FirefoxTheme.colors.textInverted,
+                    )
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text(
+                        text = stringResource(id = R.string.synced_tabs_sign_in_button),
+                        color = FirefoxTheme.colors.textInverted,
+                        fontSize = 14.sp,
+                        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+                        maxLines = 2,
+                        modifier = Modifier.align(Alignment.CenterVertically)
+                    )
+                }
+            }
+        }
+    }
+}
+
+/**
+ * UI to be displayed when a user's device has no synced tabs.
+ */
+@Composable
+fun SyncedTabsNoTabsItem() {
+    SecondaryText(
+        text = stringResource(id = R.string.synced_tabs_no_open_tabs),
+        fontSize = 12.sp,
+        maxLines = 1,
+        modifier = Modifier
+            .padding(PaddingValues(16.dp, 7.dp))
+            .fillMaxWidth()
+    )
+}
+
+@Composable
+@Preview
+private fun SyncedTabsListPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsList(
+                getFakeSyncedTabList(),
+                object : SyncedTabsView.Listener {
+                    override fun onRefresh() = Unit
+
+                    override fun onTabClicked(tab: mozilla.components.browser.storage.sync.Tab) {
+                        println("Tab clicked")
+                    }
+                }
+            )
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SyncedTabsDeviceItemPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsDeviceItem("Google Pixel Pro Max +Ultra 5000")
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SyncedTabsTabItemPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsTabItem("Mozilla", "www.mozilla.org") { println("Clicked tab") }
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SyncedTabsDeviceErrorPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsErrorItem(SyncedTabsView.ErrorType.NO_TABS_AVAILABLE)
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun SyncedTabsNoTabsItemPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer2)) {
+            SyncedTabsNoTabsItem()
+        }
+    }
+}
```

Comment:
Maybe just join them all into one Column with some Spacer?

Reply:
Mostly thinking about how to make this file smaller. Seeing that all the @Preview are defined very similarly, I think it could make sense to just display it in one view. You can imagine scenarios where you might be trying to see how it looks like with the dark theme, but adding that parameter to each @Preview compare to just one might be annoying.

Reply:
I think this makes sense. I'll condense this and reconsider previews and their brevity going forward.